app/ui/src/main/java/com/fsck/k9/activity/setup/AccountSetupBasics.java

```diff
@@ -346,16 +349,27 @@ private void onNext() {
             return;
         }
 
-        String email = mEmailView.getText().toString();
+        final String email = mEmailView.getText().toString();
         String[] emailParts = splitEmail(email);
         String domain = emailParts[1];
         mProvider = findProviderForDomain(domain);
```

Comment:
Using our own provider mechanism and Mozilla's autoconfiguration shouldn't be different code paths. They should be different implementations of the same interface. We can then loop over the different auto-config methods we support until we get a result.
The implementations themselves shouldn't be concerned with threading. The "framework" should make sure the autodiscover code runs in the background and progress messages are dispatched in the UI thread. Ideally, we'd also support canceling.

Reply:
I've added an interface and made Thunderbird one implementation, and the old providers.xml code from AccountSetupBasics has been moved to a separate class that also implements a common interface.

The "framework" class iterates over existing implementations until one of them succeeds (returns something other than `null`).